sbibm/algorithms/abcpy/rejection_abc.py

```diff
@@ -0,0 +1,141 @@
+    if sass:
+        raise NotImplementedError("SASS not yet implemented")
+
+    assert not (num_observation is None and observation is None)
+    assert not (num_observation is not None and observation is not None)
+
+    if (eps is not None) + (num_top_samples is not None) + (quantile is not None) != 1:
+        raise RuntimeError("Exactly one of `num_top_samples`, `quantile`, `eps` needs to be specified.")
+
+    log = sbibm.get_logger(__name__)
+    log.info(f"Running REJ-ABC")
+
+    if observation is None:
+        observation = task.get_observation(num_observation)
+
+    if num_top_samples is not None:
+        if sass:
+            quantile = num_top_samples / (
+                    num_simulations - int(sass_fraction * num_simulations)
+            )
+        else:
+            quantile = num_top_samples / num_simulations
+
+    # use the dummy backend only (for now)
+    backend = BackendDummy()
+
+    # as Rejection ABC works by directly drawing until you get below some threshold -> here we set the threshold to
+    # very large value and then we select the best objects later.
+
+    statistics = Identity(degree=1, cross=True)  # we assume statistics are already computed inside the task
+    distance_calc = get_distance(distance, statistics)
+
+    # define prior and model from the task simulator:
+    parameters = ABCpyPrior(task)
+    model = ABCpySimulator([parameters], task, max_calls=num_simulations)
+
+    # inference; not sure how to use random seed
+    sampler = RejectionABC(root_models=[model], distances=[distance_calc], backend=backend, seed=None)
+    # print("obs", [observation])
+    # print("obs", [np.array(observation)])
+    journal_standard_ABC = sampler.sample([[np.array(observation)]],
+                                          n_samples=num_simulations // num_simulations_per_param,
+                                          n_samples_per_param=num_simulations_per_param,
+                                          epsilon=10 ** 50)  # set epsilon to very large number
+    if eps is None:  # then we use quantile to select the posterior samples from the generated ones
+        journal_standard_ABC_reduced = journal_cleanup_rejABC(journal_standard_ABC, percentile=quantile * 100)
+    else:
+        journal_standard_ABC_reduced = journal_cleanup_rejABC(journal_standard_ABC, threshold=eps)
+
+    actual_n_simulations = journal_standard_ABC.number_of_simulations[-1] * journal_standard_ABC.configuration[
+        "n_samples_per_param"]
+    # this takes into account the fact that num_simulations can be not divisible by num_simulations_per_param
+    expected_n_simulations = (num_simulations // num_simulations_per_param) * num_simulations_per_param
+    assert actual_n_simulations == expected_n_simulations
+
+    if save_distances:
+        distances = torch.tensor(journal_standard_ABC_reduced.distances[-1])
+        save_tensor_to_csv("/home/lorenzo/Scrivania/OxWaSP/ABC-project/Code/sbibm/distances.csv", distances)
```

Comment:
We should get rid of the path here. The quick solution would be to just do `save_tensor_to_csv("distances.csv", distances)`, so saving it to the current directory, or introducing a new keyword argument

Reply:
Yes indeed, will remove that path which was for my laptop. Following the implementation in the other algorithms, I will just leave "distances.csv".